src/OrleansAzureUtils/Providers/Streams/AzureQueue/AzureQueueAdapterReceiver.cs

```diff
@@ -18,24 +19,26 @@ internal class AzureQueueAdapterReceiver : IQueueAdapterReceiver
         private long lastReadMessage;
         private Task outstandingTask;
         private readonly Logger logger;
+        private readonly bool hasLargeMessageSupport;
+
+        private readonly Dictionary<Guid, List<MessageSegment>> messageSegments = new Dictionary<Guid, List<MessageSegment>>();
 
         public QueueId Id { get; private set; }
 
-        public static IQueueAdapterReceiver Create(QueueId queueId, string dataConnectionString, string deploymentId, TimeSpan? messageVisibilityTimeout = null)
+        public static IQueueAdapterReceiver Create(QueueId queueId, string dataConnectionString, string deploymentId, bool hasLargeMesageSupport = false, TimeSpan? messageVisibilityTimeout = null)
         {
             if (queueId == null) throw new ArgumentNullException("queueId");
             if (String.IsNullOrEmpty(dataConnectionString)) throw new ArgumentNullException("dataConnectionString");
             if (String.IsNullOrEmpty(deploymentId)) throw new ArgumentNullException("deploymentId");
-            
             var queue = new AzureQueueDataManager(queueId.ToString(), deploymentId, dataConnectionString, messageVisibilityTimeout);
-            return new AzureQueueAdapterReceiver(queueId, queue);
+            return new AzureQueueAdapterReceiver(queueId, queue, hasLargeMesageSupport);
```

Comment:
How are the original CloudQueueMessages deleted from the queue in MessagesDeliveredAsync?